exalearn-api/ExaLearn.Dal/Repositories/HistoryRepository.cs

```diff
@@ -9,5 +13,15 @@ public class HistoryRepository : GenericRepository<History>, IHistoryRepository
         public HistoryRepository(ExaLearnDbContext appDbContext) : base(appDbContext)
         {
         }
+
+        public async Task<IList<History>> GetUserHistoryByIdAsync(int userId)
+        {
+            return await _appDbContext.Histories.Include(x => x.User).Include(x => x.PassedTest).Where(x => x.UserId == userId).ToListAsync();
+        }
+
+        public async Task<IList<PassedTest>> GetHRUserHistoryByIdAsync(int userId)
+        {
+            return await _appDbContext.PassedTests.Include(x => x.User).Where(x => x.UserId == userId).ToListAsync();
```

Comment:
it's the same as the line #19

Reply:
The linked data will be null, need to use eager loading .Include();